### PR TITLE
Add request retries implementation

### PR DESCRIPTION
**Pull request by fsufyan**
• Random number generator from custom seed.
• Transient error adapter for catching transient failures.
• Exponential backoff delay with full jitter implementation.
• Test cases for various conditions using a custom error mock adapter.

### CODE REVIEW

**Comment by fsufyan** [comment]
@lugehorsam Rest of client methods will be updated for retry logic after successful merging.

File: Sources/Nakama/GrpcClient.swift [hunk 1]
```diff
@@ -145,7 +146,9 @@ public class GrpcClient : Client {
         if vars != nil {
             req.account.vars = vars!
         }
-        return try await self.nakamaGrpcClient.authenticateCustom(req).response.get().toSession()
+        return try await retryInvoker.invokeWithRetry(request: {
```
Comment (lugehorsam) [inline Sources/Nakama/GrpcClient.swift:149]: are we only using the retry invoker for authenticate custom at the moment?

**Review by lugehorsam — CHANGES_REQUESTED** [review]
just one q

File: Sources/Nakama/GrpcClient.swift (hunk 1 above)
Comment (fsufyan) [reply Sources/Nakama/GrpcClient.swift:149]: This is just for the use in the tests, after we merge there will be another PR to add retry code in all other grpc methods.

Comment (lugehorsam) [reply Sources/Nakama/GrpcClient.swift:149]: ok sounds good 👍

**Review by lugehorsam — APPROVED** [review]

**Pull request merged**